src/_cffi_src/openssl/src/osrandom_engine.c

```diff
@@ -135,9 +177,7 @@ static int dev_urandom_fd(void) {
 
   error:
     if (fd != -1) {
-        do {
-            n = close(fd);
-        } while (n < 0 && errno == EINTR);
+        close(fd);
```

Comment:
This code used to assume `close` could fail and if it was `EINTR` it would retry. Is that foolish and it's safe to just assume `close` will always work/if it fails we don't care?

Reply:
Summarizing a conversation from IRC:

- On Linux the fd was generally closed if you get an EINTR, so this was wrong
- On other Unix the fd _may_ have been closed
- Signals are bad

The only time this can leak an fd now is if you use a weird unix and signals. I decided I was ok with this.

Reply:
Oh, also on linux the behavior varied between glibc and musl.